ddl/cluster.go

```diff
@@ -140,13 +143,85 @@ func checkFlashbackCluster(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job, f
 	return nil
 }
 
+func addToSlice(schema string, tableName string, tableID int64, nonFlashbackIDs, allIDs []int64) ([]int64, []int64) {
+	if filter.IsSystemSchema(schema) && !strings.HasPrefix(tableName, "stats_") {
+		nonFlashbackIDs = append(nonFlashbackIDs, tableID)
+	}
+	allIDs = append(allIDs, tableID)
+	return nonFlashbackIDs, allIDs
+}
+
+// GetFlashbackKeyRanges get keyRanges for flashback cluster
+func GetFlashbackKeyRanges(sess sessionctx.Context, startKey kv.Key) ([]kv.KeyRange, error) {
+	schemas := sess.GetDomainInfoSchema().(infoschema.InfoSchema).AllSchemas()
+
+	var keyRanges []kv.KeyRange
+	var allPhysicalIDs, nonFlashbackPhysicalIDs []int64
+	for _, db := range schemas {
+		for _, table := range db.Tables {
+			if !table.IsBaseTable() || table.ID > meta.MaxGlobalID {
```

Comment:
> table.ID > meta.MaxGlobalID

Is there any chance this might happen?

Reply:
`meta.GenGlobalID` will compare MaxGlobalID between newID. Only some ddl job tables will greater than this table.

Reply:
Got it, thanks.